app/src/main/kotlin/com/intive/picover/images/viewmodel/ImagesViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.intive.picover.images.viewmodel
+
+import android.net.Uri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.images.repository.ImagesRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ImagesViewModel @Inject constructor(private val repository: ImagesRepository) : ViewModel() {
+
+	private val _uiState = MutableStateFlow<LatestFirebaseStorageState>(LatestFirebaseStorageState.Loading)
```

Comment:
probably good topic to discuss, if we want to use underscore in the property name. I see the reason why it was used, but i think it is something worth discussion :)

Reply:
I agree.

Reply:
We will implement a new approach for backing fields when it's going to be available with the new version of Kotlin.  
[to do issue](https://github.com/orgs/intive/projects/3/views/1?pane=issue&itemId=26197144)